plugins/source/okta/codegen/recipes/applications.go

```diff
@@ -0,0 +1,64 @@
+package recipes
+
+import (
+	"github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/okta/okta-sdk-golang/v2/okta"
+)
+
+func Applications() []*Resource {
+	return []*Resource{
+		{
+			DataStruct: &okta.Application{},
+			PKColumns:  []string{"id"},
+			Service:    "applications",
+			SkipFields: []string{"Profile"},
+			ExtraColumns: []codegen.ColumnDefinition{
+				{
+					Name:     "profile",
+					Type:     schema.TypeJSON,
+					Resolver: `schema.PathResolver("Profile")`,
+				},
+			},
+			Relations: []string{"ApplicationUsers()", "ApplicationGroupAssignments()"},
+		},
+		{
+			DataStruct: &okta.AppUser{},
+			PKColumns:  []string{"app_id", "id"},
```

Comment:
instead of adding `app_id` here you could add the PK option right in the extra col def

Reply:
better yet, utilize the following func
```go
func parentID(pfx string) codegen.ColumnDefinition {
	const parentIdResolver = `schema.ParentColumnResolver("id")`
	return codegen.ColumnDefinition{
		Name:     pfx + "_id",
		Type:     schema.TypeString,
		Resolver: parentIdResolver,
		Options:  schema.ColumnCreationOptions{PrimaryKey: true},
	}
}
```
and use it

Reply:
I find it more self contained this way, all PKs are in a single line.

Reply:
Yes, but the order will not be preserved, so you still need to be careful.
it might be a good thing to think about pk columns enforcing the pk order, though
@yevgenypats

Reply:
I don't think it's important atm and don't have time to review this throughly for a small plugin with little usage so whatever @disq is going for.